testsuite/features/step_definitions/common_steps.rb

```diff
@@ -239,18 +239,29 @@
 
 When(/^I remove kickstart profiles and distros$/) do
   host = $server.full_hostname
+  # -------------------------------
+  # Cleanup kickstart distros and their profiles, if any.
   @client_api = XMLRPC::Client.new2('http://' + host + '/rpc/api')
   @sid = @client_api.call('auth.login', 'admin', 'admin')
-  # -------------------------------
-  # cleanup kickstart profiles and distros
-  distro_name = 'fedora_kickstart_distro'
-  @client_api.call('kickstart.tree.delete_tree_and_profiles', @sid, distro_name)
+
+  # Get all distributions: created from UI or from XMLRPC API.
+  distros = $server.run('cobbler distro list')[0].split
+
+  # The name of distros created in the UI has the form: distro_label + suffix
+  user_details = @client_api.call('user.get_details', @sid, 'testing')
```

Comment:
It might be good to use `XMLRPCUserTest` class instead of this generic one, but, I'm also ok with that approach.